javascript/channel.ts

```diff
@@ -55,10 +55,19 @@ export class Channel {
                     this.readBuf = this.readBuf.slice(len);
                     resolve(data);
                     if (this.readBuf.length == 0 && this.gotEOF) {
+                        console.log("gotEOF, setting to undefined");
                         this.readBuf = undefined;
                     }
                     return;
                 }
+                if (this.gotEOF) {
+                    console.log("gotEOF, resolving rest of data");
+                    // TODO what if length == 0, return undefined then too?
+                    let data = this.readBuf;
+                    this.readBuf = undefined;
+                    resolve(data);
+                    return;
+                }
```

Comment:
here if we got EOF and it's less than the requested `len`, return it and clear the buffer